core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -207,7 +207,7 @@ struct HIPParallelLaunch<DriverType, Kokkos::LaunchBounds<0, 0>,
       HIP_SAFE_CALL(hipMemcpyAsync(d_driver, &driver, sizeof(DriverType),
                                    hipMemcpyHostToDevice,
                                    hip_instance->m_stream));
-      hip_parallel_launch_local_memory<DriverType>
+      hip_parallel_launch_local_memory<DriverType, 1024, 1>
```

Comment:
uhm that is weird guys. You are saying for this now that the launch HAS to work with block size of 1024. And thus you HAVE to be below a certain register limit.

Reply:
At the very least, please don't put large unnamed constants in your code without at least a comment explaining them.